ts/annotations/annotations.src.ts

```diff
@@ -1047,12 +1050,38 @@ merge(
          * @private
          */
         init: function (this: Highcharts.Annotation): void {
+            const plotOptions = this.chart.options.plotOptions as any,
+                defer = this.options.defer;
+
             this.linkPoints();
             this.addControlPoints();
             this.addShapes();
             this.addLabels();
             this.addClipPaths();
             this.setLabelCollector();
+            this.deferTime = (defer === false || this.chart.renderer.forExport) ?
+                // If defer is disabled deferTime is set to 0
+                // (invoke immediately)
+                0 : (
+                    isNumber(defer) ?
+                        // If defer is a number - animate will triger
+                        // after this number, otherwise the time will be taken
+                        // from the plotOptions.series.animation
+                        defer :
+                        (plotOptions.series && defined(plotOptions.series.animation) ?
+                            (
+                                (
+                                    plotOptions.series.animation.defer &&
+                                    plotOptions.series.animation.duration
+                                ) ?
+                                // If defer and duration are set the animation
+                                //  will be triggered after their sum value
+                                    plotOptions.series.animation.defer + plotOptions.series.animation.duration :
+                                    plotOptions.series.animation.defer ? plotOptions.series.animation.defer :
```

Comment:
If I am counting correctly this is the 5th level of the nested ternary operators. Readability is low. It would be nice to see this rewritten into something maintainable more easily.

Reply:
@KacperMadej please check the current implementation and let me know what do you think about it.

Reply:
Thnak you :+1: - it's easier to follow the code now.

whispers: _(You could also try to eliminate the any casting and set the default value (from the last else) at the top (when the variable is created), but those are just minor details and really NOT worth taking care of right now if the code is not final.)_